base-trezor-keyring.js

```diff
@@ -0,0 +1,562 @@
+    throw new Error('Method init not implemented');
+  }
+
+  /**
+   * Gets the model, if known.
+   * This may be `undefined` if the model hasn't been loaded yet.
+   *
+   * @returns {"T" | "1" | undefined}
+   */
+  getModel() {
+    return this.model;
+  }
+
+  /**
+   * Clears the iframe created by Trezor
+   *
+   * @returns {Promise<void>}
+   */
+  dispose() {
+    throw new Error('Method dispose not implemented');
+  }
+
+  /**
+   * Gets public key from device
+   *
+   * @returns {Promise}
+   */
+  _getPublicKey(_payload) {
+    throw new Error('Method _getPublicKey not implemented');
+  }
+
+  /**
+   * Signs transaction using the device
+   *
+   * @returns {Promise}
+   */
+  _ethereumSignTransaction(_payload) {
+    throw new Error('Method _ethereumSignTransaction not implemented');
+  }
+
+  /**
+   * Signs message using the device
+   *
+   * @returns {Promise}
+   */
+  _ethereumSignMessage(_payload) {
+    throw new Error('Method _ethereumSignMessage not implemented');
+  }
+
+  /**
+   * Signs typed data using the device
+   *
+   * @returns {Promise}
+   */
+  _ethereumSignTypedData(_payload) {
+    throw new Error('Method _ethereumSignTypedData not implemented');
+  }
+
+  serialize() {
+    return Promise.resolve({
+      hdPath: this.hdPath,
+      accounts: this.accounts,
+      page: this.page,
+      paths: this.paths,
+      perPage: this.perPage,
+      unlockedAccount: this.unlockedAccount,
+    });
+  }
+
+  deserialize(opts = {}) {
+    this.hdPath = opts.hdPath || hdPathString;
+    this.accounts = opts.accounts || [];
+    this.page = opts.page || 0;
+    this.perPage = opts.perPage || 5;
+    return Promise.resolve();
+  }
+
+  isUnlocked() {
+    return Boolean(this.hdk && this.hdk.publicKey);
+  }
+
+  unlock() {
+    if (this.isUnlocked()) {
+      return Promise.resolve('already unlocked');
+    }
+    return new Promise((resolve, reject) => {
+      this._getPublicKey({
+        path: this.hdPath,
+        coin: 'ETH',
+      })
+        .then((response) => {
+          if (response.success) {
+            this.hdk.publicKey = Buffer.from(response.payload.publicKey, 'hex');
+            this.hdk.chainCode = Buffer.from(response.payload.chainCode, 'hex');
+            resolve('just unlocked');
+          } else {
+            reject(
+              new Error(
+                (response.payload && response.payload.error) || 'Unknown error',
+              ),
+            );
+          }
+        })
+        .catch((e) => {
+          reject(new Error((e && e.toString()) || 'Unknown error'));
+        });
+    });
+  }
+
+  setAccountToUnlock(index) {
+    this.unlockedAccount = parseInt(index, 10);
+  }
+
+  addAccounts(n = 1) {
+    return new Promise((resolve, reject) => {
+      this.unlock()
+        .then((_) => {
+          const from = this.unlockedAccount;
+          const to = from + n;
+
+          for (let i = from; i < to; i++) {
+            const address = this._addressFromIndex(pathBase, i);
```

Comment:
```suggestion
          for (let i = 0; i < n; i++) {
            const address = this._addressFromIndex(pathBase, this.unlockedAccount + i);
```
Are the `from`/`to` bookkeeping variables really helping? I find this more clear.